### ISSUE TITLE
PyLint's useless-suppression is not producing consistent behaviour with `mock.patch.multiple_targets`

### ISSUE THREAD

**Issue by nbro10** [pylint-dev/pylint#8977]
### Bug description

So, in my tests, I have something like this

```python
from unittest import mock

def whatever():
    mocks = mock.patch.multiple_targets(  # pylint: disable=no-member
        [
            mock.patch(MODULE_PATH + '.func1),
            mock.patch(MODULE_PATH + '.func2'),
        ]
    )
```

If I run pylint with the suppression enabled, **sometimes**, it tells me that this disable is useless. So, I tried to remove it, but then I always get the error that `multiple_targets` is not a member

```
E1101: Function 'patch' has no 'multiple_targets' member (no-member)
```

The weirdest thing is that the useless suppression warning doesn't always happen or happens for different files

### Configuration

```ini
Python 3.8.13
```

### Command used

```shell
poetry run pylint my_package tests docs scripts --enable=useless-suppression
```

### Pylint output

```shell
E1101: Function 'patch' has no 'multiple_targets' member (no-member)

Or

I0021: Useless suppression of 'no-member' (useless-suppression)
```

### Expected behavior

PyLint is consistent

### Pylint version

```shell
PyLint 2.14.5
Python 3.8.13
Astroid 2.11.7
```

### OS / Environment

Mac OS Monterey, Apple M1

**Comment by Pierre-Sassoulas** [comment]
By **sometime** do you mean in the same environment when calling pylint multiple time ? Or are you using multiple interpreters or multiple environments (differing dependencies, OS, etc) ? Also you could upgrade pylint to 3.0.0a7 to see if the problem still exists :)

**Comment by Pierre-Sassoulas** [comment]
So if I understand you well you launch once with `` --enable=useless-suppression`` and the other without ? The code you gave is not parseable (``a.py:7:38: E0001: Parsing failed: 'unterminated string literal (detected at line 7) (<unknown>, line 7)' (syntax-error)``) so I parsed the following using both option:
```python
from pathlib import Path
from unittest import mock

MODULE_PATH = Path(__file__)

def whatever():
    mocks = mock.patch.multiple_targets(  # pylint: disable=no-member
        [
            mock.patch(MODULE_PATH + '.func1'),
            mock.patch(MODULE_PATH + '.func2'),
        ]
    )
```
I don't have any issues on main / python 3.11 with this snippet. We did change a lot of things since 2.14, and some for useless-suppression (maybe https://github.com/pylint-dev/pylint/pull/7887 in 2.15, but there might be others), so it's probable it was fixed on main and sadly we don't support python 3.7 anymore. Please reopen an issue if you have a reproducer for more recent version of pylint.
